components/brave_wallet/browser/simulation_responses.idl

```diff
@@ -0,0 +1,270 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Dummy comments to make the linter happy.
+namespace simulation_responses {
```

Comment:
Sure! We have both an IDL as well as Mojo interfaces, but they serve different purposes.

The IDL describes the JSON schema and is translated into a generated C++ file by `tools/json_schema_compiler`. The mojo file describes types that will be consumed by the clients. The process flow below should make it clear how everything fits.

```
JSON string ---(1)---> base::Value ---(2)---> types generated from IDL ---(3)---> mojo types

(1) APIRequestResult::value_body()
(2) simulation_responses::SolanaSimulationResponse::FromValue(...)
(3) code in simulation_response_parser.cc
```

FYI, we started following the IDL approach since https://github.com/brave/brave-core/pull/15999.

Reply:
The main goal of using IDL for parsing JSON response is to eliminate redundant and sometimes error prone codes which are replaced by generated codes.

Reply:
Cool, thanks for the context here.